### PR TITLE
[0-size Tensor No.156] Add 0-size Tensor support for paddle.nn.functional. binary_cross_entropy_with_logits [fluid_ops]

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Operator Mechanism

### PR Types

Bug fixes

### Description

156 paddle.nn.functional.binary_cross_entropy_with_logits
PaddleAPITest 测试通过增加单测
image

使用full创建的tensor，没有反向
https://github.com/PaddlePaddle/Paddle/blob/f53683b0d6e85ccddeab240ddc2918fc78ac922e/python/paddle/nn/functional/loss.py#L845
image

sigmoid_cross_entropy_with_logits XPU kernel CI有错误，修改XPU kernel

### CODE REVIEW

**Review by DanielSun11 — APPROVED** [review]
LGTM
